video/out/wayland_common.c

```diff
@@ -3489,8 +3492,19 @@ static void set_color_management(struct vo_wayland_state *wl)
         wp_image_description_creator_params_v1_set_max_cll(image_creator_params, lrintf(hdr.max_cll));
         wp_image_description_creator_params_v1_set_max_fall(image_creator_params, lrintf(hdr.max_fall));
     }
+
     struct wp_image_description_v1 *image_description = wp_image_description_creator_params_v1_create(image_creator_params);
+    cm_queue = wl_display_create_queue(wl->display);
+    wl_proxy_set_queue((struct wl_proxy *)image_description, cm_queue);
+    wl->image_desc_done = false;
     wp_image_description_v1_add_listener(image_description, &image_description_listener, wl);
+
+    /* Block here to ensure the compositor is ready to receive the
+     * image_description before we do anything else. */
+    while (wl->image_desc_done == false)
+        wl_display_dispatch_queue(wl->display, cm_queue);
```

Comment:
Hmm not 100% sure but isn't it more conceptually correct to process events like what `wayland_dispatch_events` does?

Reply:
What do you mean specifically?

Reply:
we can actually use wl_display_roundtrip_queue here. I previously didn't want to use it because technically `roundtrip` implies that we must wait until the compositor processes all events, but since we're making a new queue here the only event will be image_description anyway.

Reply:
I was thinking you might need to `wl_display_prepare_read_queue` and all that. But `wl_display_roundtrip_queue` sounds like the right solution.